test/cpp/jit/test_gpu.cpp

```diff
@@ -3066,80 +3066,188 @@ void testGPU_FusionSimpleGemm() {
   }
 }
 
-// This test currently requires a combination of broadcast and reduction
-// operations and parellelization strategy that is currently not supported.
-// It is a goal to get this example working and this test is added so we
-// can continue working on getting this example fixed. Right now it
-// produces an incorrect result. Either we need to error coherently on the
-// optimization strategy we don't support and set this test to one we do support
-// or we need to get this schedule working correctly.
+// Softmax without input normalization. This is a simplified version
+// of testGPU_FusionSoftmaxNormalized.
 void testGPU_FusionSoftmax() {
   torch::jit::fuser::cuda::CudaKernel prog;
   Fusion& fusion = *prog.fusion_;
   FusionGuard fg(&fusion);
 
+  const int tidx = 32;
+  const int dimx = 32;
+  const int dimy = 16;
+  const int dimz = 130;
+
   // Set up your input tensor views
   TensorView* input_tv0 = makeDummyTensor(3);
   fusion.addInput(input_tv0);
 
+  TensorView* exp_tv4 = unaryOp(UnaryOpType::Exp, input_tv0);
+  TensorView* sum_exp_tv5 =
+      reductionOp(BinaryOpType::Add, {2}, new Float(0), exp_tv4);
+  TensorView* bcast_sum_tv6 = broadcast(sum_exp_tv5, {false, false, true});
+
+  // Replicate exp_tv4 as exp_tv4_2 because exp_tv4 is going to be
+  // computed at sum_exp_rf_tv8.
+  TensorView* exp_tv4_2 = unaryOp(UnaryOpType::Exp, input_tv0);
+
+  TensorView* output_tv6 = div(exp_tv4_2, bcast_sum_tv6);
+
+  fusion.addOutput(output_tv6);
+
+  sum_exp_tv5->split(-1, tidx);
+  TensorView* sum_exp_rf_tv8 = sum_exp_tv5->rFactor({-2});
+
+  exp_tv4->split(-1, tidx);
+  exp_tv4_2->split(-1, tidx);
+  output_tv6->split(-1, tidx);
+
+  exp_tv4->computeAt(sum_exp_rf_tv8, -1);
+  exp_tv4_2->computeAt(output_tv6, -1);
+
+  exp_tv4->axis(0)->parallelize(ParallelType::BIDx);
```

Comment:
just curious, isn't enough to parallelize just the final output? (and let computeAt do the rest)

Reply:
That's probably right. The explicit style of adding parallelization specifications also comes from the way the test was originally written.

Reply:
Done